cli/src/commands/extensions/api.rs

```diff
@@ -0,0 +1,175 @@
+//! Most functions in this module are marked `#[allow(unused)]` for the time being. This
+//! intentional, as there are no clients for those functions in the rest of the code, but
+//! those functions will have to be used by the Deno integration; at that point, we may
+//! remove the annotations.
+
+use std::path::Path;
+use std::str::FromStr;
+
+use crate::commands::parse::{get_packages_from_lockfile, LOCKFILE_PARSERS};
+use crate::config::{get_current_project, Config};
+use crate::{api::PhylumApi, auth::UserInfo};
+
+use anyhow::{anyhow, Context, Error, Result};
+use deno_core::OpState;
+use once_cell::sync::Lazy;
+use phylum_types::types::auth::{AccessToken, RefreshToken};
+use phylum_types::types::common::{JobId, ProjectId};
+use phylum_types::types::job::JobStatusResponse;
+use phylum_types::types::package::{
+    Package, PackageDescriptor, PackageStatusExtended, PackageType,
+};
+use phylum_types::types::project::ProjectDetailsResponse;
+
+/// Container for lazily evaluated dependencies of Extensions API functions. These values won't be
+/// visible from extension code and will have to be set up by the JS runtime builder which will
+/// load their configuration as any other command and then provide the pertinent factories.
+struct InjectedDependencies {
+    api: Lazy<PhylumApi>,
+    config: Lazy<Config>,
+}
```

Comment:
Instantiating things like `PhylumApi` is expensive as it touches network and not every extension may need to do that, but we are going to have to provide these values ahead of time (via `OpState`). Having those lazy means we don't pay the cost upfront.

Reply:
I feel like it's going to be much simpler to just always create the `PhylumApi`. Should there be any performance concerns, which would obviously also affect our own subcommands where we don't do this, we can optimize in the future.

Reply:
We don't initialize `PhylumApi` for each of the CLI's commands. On my VM there is a latency of a couple seconds for commands that do use the API compared to the ones that don't, which is high enough in my opinion to warrant opting into it only when necessary (granted, it might depend on my specific networking configuration, but we should not assume the happy path for all the users).

Reply:
Yeah fair enough, there certainly is some delay at the very least. And another point is the authentication that might be required when the user is not signed in already.

Though I believe that would probably look pretty wonky when done lazily? We wouldn't want to interrupt extension's CLI output with our CLI's auth messages.